toggl_webhooks.py

```diff
@@ -41,21 +46,30 @@ async def main():
 # Utility method to generate the handler. This is necessary because we want to pass method and url
 # in a closure, and we can't use a lambda because TogglClient expects an async handler (which lambdas don't
 # support)
-def generate_message_handler(method: str, url: str):
+def generate_message_handler(method: str, url: str, **kwargs):
     async def message_handler_outer(action, model, msg: TogglSocketMessage):
-        await message_handler(action, model, method, url, msg)
-        return
+        await message_handler(action, model, method, url, msg, **kwargs)
 
     return message_handler_outer
 
 
 # The actual handler; in this case, it just sends the data to the server in the manner defined by
 # the config file.
-async def message_handler(action: str, model: str, method: str, url: str, msg: TogglSocketMessage):
-    res = requests.request(method, url, data=msg.to_dict())
-    print(f'{action} {model} -> {res.status_code} {method} {url}')
-
-    return
+async def message_handler(action: str, model: str, method: str, url: str, msg: TogglSocketMessage,
+                          trials=None, **request_kwargs):
+    trials = trials or 1
+    assert trials > 1
```

Comment:
@ihoru Shouldn't this be `assert trials >= 1`? If someone configures only one trial, or leave out the optional parameter, the assertion will fail

Reply:
Oh, yeah, sure!